libraries/azure_storage_account.rb

```diff
@@ -143,6 +145,144 @@ def table_properties
     end
   end
 
+  def blobs_diagnostic_settings
+    return unless exists?
+    # `additional_resource_properties` method will create a singleton method with the `property_name`
+    # and make api response available through this property.
+    additional_resource_properties(
+      {
+        property_name: 'diagnostic_settings',
+        property_endpoint: "#{id}/blobServices/default/providers/microsoft.insights/diagnosticSettings",
+        api_version: @opts[:diagnostic_settings_api_version],
+      },
+    )
+  end
+
+  def tables_diagnostic_settings
+    return unless exists?
+    # `additional_resource_properties` method will create a singleton method with the `property_name`
+    # and make api response available through this property.
+    additional_resource_properties(
+      {
+        property_name: 'diagnostic_settings',
+        property_endpoint: "#{id}/tableServices/default/providers/microsoft.insights/diagnosticSettings",
+        api_version: @opts[:diagnostic_settings_api_version],
+      },
+    )
+  end
+
+  def queues_diagnostic_settings
+    return unless exists?
+    # `additional_resource_properties` method will create a singleton method with the `property_name`
+    # and make api response available through this property.
+    additional_resource_properties(
+      {
+        property_name: 'diagnostic_settings',
+        property_endpoint: "#{id}/queueServices/default/providers/microsoft.insights/diagnosticSettings",
+        api_version: @opts[:diagnostic_settings_api_version],
+      },
+    )
+  end
+
+  def has_blobs_read_log_enabled?
+    return false if blobs_diagnostic_settings.nil? || blobs_diagnostic_settings.empty?
```

Comment:
inspec now comes shipped with `activesupport`, so we can use 

```suggestion
    return false if blobs_diagnostic_settings.blank?
```

Reply:
@sathish-progress , thanks for the review. Made the necessary changes